cmd/relay-gossip/client/client.go

```diff
@@ -26,7 +28,19 @@ type Client struct {
 	}
 }
 
-// NewWtihPubsub creates a gossip randomness client.
+// WithPubsub provides an option for integrating pubsub notification
+// into a drand client.
+func WithPubsub(ps *pubsub.PubSub, networkName string) dclient.Option {
+	return dclient.WithWatcher(func(_ *key.Group) (dclient.Watcher, error) {
```

Comment:
note that by not taking in the group here, no validation of results is occuring. dunno if we want this watcher to internalize that responsibility, or if there's a higher level that should do the filtering. having it happen in here means we don't have to worry about caching bad results or not falling back when we should.

for reference: https://github.com/drand/drand/blob/master/client/http.go#L137-L149

Reply:
I think it is OK to leave it without group info for now. One of the design ideas that @alanshaw introduced in his code is to have the network name be the group hash, which seems like a good way to go.

Reply:
agree that group hash as topic is a good addition. I think each member of the gossip network should be validating received randomness messages before rebroadcasting them. Agree that's a separate task from this PR though.